### DISCUSSION TITLE
Running blockcheck.sh for NS_BINDING_ABORTED queries

### DISCUSSION THREAD

**Discussion by Hi-Angel** [bol-van/zapret#336] [General]
This may be related to [this thread](https://github.com/bol-van/zapret/discussions/new?category=general), but I figured it's a specific question, so worth a separate post.

While trying to figure out problems with music.youtube.com on Moscow RosTelecom, I found on <kbd>F12</kbd> network tab that many requests fail with `NS_BINDING_ABORTED`.

![image](https://github.com/user-attachments/assets/e01e483f-3539-4426-aef2-e676507328a7)

The following option helps a lot *(assuming nfqws is enabled)*:

```
NFQWS_OPT_DESYNC_HTTPS="--dpi-desync=split --dpi-desync-ttl=0 --dpi-desync-fooling=badsum"
```

However, though less frequently, but I still get `NS_BINDING_ABORTED` *(which results I presume in playback stopping sometimes)*. Experimenting with other options yielded nothing useful, so am wondering if it's possible to run `blockcheck.sh` checks on a query that can be copied from the browser? Note that the domain is accessible with `netcat`, so checks has to be run on a query and not a domain.

**Comment by bol-van** [comment]
You can't just curl these URLs. It's POST not GET and you would have to reproduces cookies and POST data. It may also expire fast.
If you want to detect failures during download phase you have to modify blockcheck source. It's not supported from the box.
Also it cannot measure speed

**Reply by Hi-Angel** [reply]
> You can't just curl these URLs. It's POST not GET and you would have to reproduces cookies and POST data

Actually, Firefox does provide a "Copy Value as cURL" option, which includes cookies and all the necessary stuff. When I execute such curl command it just times out *(which I guess is what `NS_BINDING_ABORTED` error is about)*.

**Reply by bol-van** [reply]
Thanx, I didn't know about that.
You still have to temporary-hack blockcheck source
for example here : curl_with_subst_ip
I could not just copy-paste the command there. Shell complains about ( ) chars escapement

**Reply by Hi-Angel** [reply]
Thank you, editing `curl_with_subst_ip` to replace this string `${arg:+$arg }$connect_to` with curl-parameters I got from Firefox helped to solve the problem. This is the output:

```
ipv4 rutracker.org curl_test_https_tls12 : tpws not working
ipv4 rutracker.org curl_test_https_tls12 : nfqws --dpi-desync=fake --dpi-desync-ttl=6
ipv4 rutracker.org curl_test_http3 : nfqws --dpi-desync=fake
```

*(note: this is unrelated to rutracker, because I hardcoded the destination to whatever I've gotten from Firefox)*.

Youtube music now works much better.

**Reply by ovprc** [reply]
Спасибо, теперь я точно знаю, что у меня вообще нету обхода под ютуб🙂
Челябинск. Провайдер "Интерсвязь". (Локальный, городской)
Провайдер подменяет DNS, пользуюсь DNSCrypt.

Потратил две недели с головой, вчитываясь все во все дискуссии, иссуе и ридми, по итогу щас прогнал написанный метод выше - какой то процент запросов перенаправило на провайдера, другой процент выбило в bad request от клаудфлеера, а остальная часть просто выдала ноль байтов.

Перебрал тонны вариаций отсюда, как и со своего blockcheck. Ничего не обошло замедление ютуба, однако для всех остальных сайтов хватило настройки:
`--wf-tcp=443 --dpi-desync=fake --dpi-desync-fooling=md5sig`
(Перебирал и на линуксе, и на винде - результат один и тот же.)

Продолжаю отчаиваться с юбустом на борту
